arviz/tests/base_tests/test_data.py

```diff
@@ -323,6 +324,18 @@ def test_addition(self):
         fails = check_multiple_attrs(test_dict, new_idata)
         assert not fails
 
+    def test_extend_xr_method(self):
+        arr = xr.DataArray(
+            data=np.ones((2, 3)),
+            dims=["x", "y"],
+            coords={"x": range(2), "y": range(3), "a": ("x", [3, 4])},
+        )
+        ds = xr.Dataset({"v": arr})
```

Comment:
I think it will be easier to initialize `ds` with `from_dict`, especially because I think that the created inference data should have several groups, otherwise we cannot test that the method is applied to all desired groups and **not** applied to the excluded groups. For example with 3 groups, the method can be applied to two of them and not to the third, and there are not too much groups to test each of the 3 manually. Having said that, maybe it makes more sense to create a helper function to compare an inference data to a list of datasets or something like this. I can try to think about some useful api for the helper.

Reply:
Yeah, I am planning to add a `equals` method to InferenceData, similar to `xarray's` one.